tests/PhpWordTests/Writer/Word2007/Element/TableTest.php

```diff
@@ -0,0 +1,218 @@
+ */
+class TableTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * Executed after each method of the class.
+     */
+    protected function tearDown(): void
+    {
+        TestHelperDOCX::clear();
+    }
+
+    public static function testTableNormal(): void
+    {
+        $phpWord = new PhpWord();
+        $section = $phpWord->addSection();
+        $section->addText('Before table (normal).');
+        $table = $section->addTable(['width' => 5000, 'unit' => TblWidth::PERCENT]);
+        $row = $table->addRow();
+        $tc = $table->addCell();
+        $tc->addText('R1C1');
+        $tc = $table->addCell();
+        $tc->addText('R1C2');
+        $row = $table->addRow();
+        $tc = $table->addCell();
+        $tc->addText('R2C1');
+        $tc = $table->addCell();
+        $tc->addText('R2C2');
+        $row = $table->addRow();
+        $tc = $table->addCell();
+        $tc->addText('R3C1');
+        $tc = $table->addCell();
+        $tc->addText('R3C2');
+        $section->addText('After table.');
+
+        $doc = TestHelperDOCX::getDocument($phpWord);
+        self::assertFalse($doc->elementExists('/w:document/w:body/w:tbl[2]'), 'should be only 1 table');
+
+        self::assertTrue($doc->elementExists('/w:document/w:body/w:tbl/w:tr[1]'));
+        self::assertTrue($doc->elementExists('/w:document/w:body/w:tbl/w:tr[1]/w:tc'));
+        self::assertTrue($doc->elementExists('/w:document/w:body/w:tbl/w:tr[1]/w:tc[2]'));
+        self::assertFalse($doc->elementExists('/w:document/w:body/w:tbl/w:tr[1]/w:tc[3]'));
+
+        self::assertTrue($doc->elementExists('/w:document/w:body/w:tbl/w:tr[2]'));
+        self::assertTrue($doc->elementExists('/w:document/w:body/w:tbl/w:tr[2]/w:tc'));
+        self::assertTrue($doc->elementExists('/w:document/w:body/w:tbl/w:tr[2]/w:tc[2]'));
+        self::assertFalse($doc->elementExists('/w:document/w:body/w:tbl/w:tr[2]/w:tc[3]'));
+
+        self::assertTrue($doc->elementExists('/w:document/w:body/w:tbl/w:tr[3]'));
+        self::assertTrue($doc->elementExists('/w:document/w:body/w:tbl/w:tr[3]/w:tc'));
+        self::assertTrue($doc->elementExists('/w:document/w:body/w:tbl/w:tr[3]/w:tc[2]'));
+        self::assertFalse($doc->elementExists('/w:document/w:body/w:tbl/w:tr[3]/w:tc[3]'));
+    }
+
+    public static function testSomeRowWithNoCells(): void
+    {
+        $phpWord = new PhpWord();
+        $section = $phpWord->addSection();
+        $section->addText('Before table (row 2 has no cells).');
+        $table = $section->addTable(['width' => 5000, 'unit' => TblWidth::PERCENT]);
+        $row = $table->addRow();
+        $tc = $table->addCell();
+        $tc->addText('R1C1');
+        $tc = $table->addCell();
+        $tc->addText('R1C2');
+        $row = $table->addRow();
+        /*
+        $tc = $table->addCell();
+        $tc->addText('R2C1');
+        $tc = $table->addCell();
+        $tc->addText('R2C2');
+        */
```

Comment:
```suggestion
        // Add no cell to the current row
```

Reply:
Done. Tests were basically cloned from each other. It was helpful during development to just comment out what differed from test to test. Agree that's not needed now.